pkg/upgradeconfigmanager/upgradeconfigmanager.go

```diff
@@ -245,18 +246,22 @@ func (s *upgradeConfigManager) Refresh() (bool, error) {
 	// is there a difference between the original and replacement?
 	changed := !reflect.DeepEqual(replacementUpgradeConfig.Spec, currentUpgradeConfig.Spec)
 	if changed {
-		// Apply the resource
-		log.Info("cluster upgrade spec has changed, will update")
-		err = s.client.Update(context.TODO(), &replacementUpgradeConfig)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				// couldn't update because it didn't exist - create it instead.
-				err = s.client.Create(context.TODO(), &replacementUpgradeConfig)
+		if foundUpgradeConfig {
+			// Delete and re-create the resource
+			log.Info("cluster upgrade spec has changed, will delete and re-create.")
+			err = s.client.Delete(context.TODO(), currentUpgradeConfig)
```

Comment:
Can you set `--grace-period` and `--force` in the options? I have found deletion can be tricky as it marks a resource for deletion but can take some time.
You might hit errors on recreation if its not gone

Reply:
Yeah that's a good point. 
I think looking at `DeletePropagationForeground` is the preferred way here. https://github.com/ericchiang/k8s/blob/08b7bf46703ad24b5420601bf6df6d69b4b74201/resource.go#L72

I'm also not sure if we could potentially face resource version conflict errors - because this operates in a separate goroutine to the UC controller, you could potential have a situation where the UC controller updates the UpgradeConfig in-between the time that the UC Manager reads the UpgradeConfig and deletes it. (but again, not sure if deletes are subject to this)

Reply:
Resolved w/ https://github.com/openshift/managed-upgrade-operator/pull/181/commits/44f0d1fe3d7e0af222b2e8dd86fd26ba635f4a63